### PR TITLE
Feature/273 20 implement constraints on backend fixed

### PR DESCRIPTION
**Pull request by Ashalynd**
A replacement for https://github.com/StarterSquad/Optima/pull/336 which cannot be merged.

### CODE REVIEW

**Comment by Ashalynd** [comment]
@cliffckerr @robynstuart error in cost-coverage (I created new project, ran auto-calibrate and then went to cost-coverage / FWS and started filling in fields for the main graph)

```
Traceback (most recent call last):
  File "/Users/anna/git/Optima/server/src/optima/model.py", line 359, in doCostCoverage
    plotdata, plotdata_co, plotdata_cc, effectnames, D = plotallcurves(**args)
  File "/Users/anna/git/Optima/server/src/sim/makeccocs.py", line 456, in plotallcurves
    plotdata_cc, D = makecc(D=D, progname=progname, ccplot=ccplot, ccparams=ccparams, verbose=verbose)
  File "/Users/anna/git/Optima/server/src/sim/makeccocs.py", line 91, in makecc
    coverage, targetpopsize, coveragelabel, convertedccparams, ccplottingparams = getcoverage(D, ccparams, popadj=popadj, artelig=default_artelig, progname=progname)
  File "/Users/anna/git/Optima/server/src/sim/makeccocs.py", line 578, in getcoverage
    growthratel = (-1/params[3])*log((2*params[0])/(params[1]+params[0]) - 1)
TypeError: unsupported operand type(s) for /: 'int' and 'NoneType'
```

**Comment by robynstuart** [comment]
@Ashalynd yes, it needs another parameters to be added on FE - see https://trello.com/c/d3cZDhoV/303-add-uncertainties-to-ccocs. Not sure what happened on BE - what test were you running?

File: client/source/js/modules/model/cost-coverage-ctrl.js [hunk 1]
```diff
@@ -176,19 +177,32 @@ define(['./module', 'underscore'], function (module, _) {
      * Generates ready to plot graph for a cost coverage.
      */
     var prepareCostCoverageGraph = function (data) {
+      console.log("cost-coverage data", data);
```
Comment (nikgraf) [inline client/source/js/modules/model/cost-coverage-ctrl.js]: please remember to check your PRs for debugging logs

Comment (Ashalynd) [inline client/source/js/modules/model/cost-coverage-ctrl.js]: that was the one I missed

File: client/source/js/modules/model/cost-coverage-ctrl.js [hunk 2]
```diff
@@ -176,19 +177,32 @@ define(['./module', 'underscore'], function (module, _) {
...
 
-      _(data.xlinedata).each(function (x, index) {
-        var y = data.ylinedata;
-        graph.data.lines[0].push([x, y[index]]);
-      });
+      if (data.ylinedata) {
+        var numOfLines = data.ylinedata.length;
+        _(data.xlinedata).each(function (x, index) {
+          var y = data.ylinedata;
+          for (var i = 0; i < numOfLines; i++) {
```
Comment (nikgraf) [inline client/source/js/modules/model/cost-coverage-ctrl.js]: just some code sugar: here we could use underscore's each with index as well

**Pull request merged**